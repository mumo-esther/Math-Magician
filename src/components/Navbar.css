* {
  box-sizing: border-box;
}

body {
  margin: 0;
}

.nav {
  margin-top: 20px;
  color: blue;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 2rem;
  padding: 0 1rem;
}

.nav ul {
  padding: 0;
  margin: 0;
  list-style: none;
  display: flex;
  justify-content: center;
  gap: 1rem;
}

.nav a {
  color: inherit;
}

.title {
  font-size: 40px;
  color: black;
  font-weight: 900;
  text-decoration-line: none;
}
