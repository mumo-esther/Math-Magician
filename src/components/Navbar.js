import { Link, useMatch, useResolvedPath } from 'react-router-dom';
import React from 'react';
import PropTypes from 'prop-types';
import './Navbar.css';

const Navbar = () => (
  <nav className="nav">
    <Link to="/" className="title">
      Math Magicians
    </Link>
    <ul>
      <CustomLink to="/Home">Home</CustomLink>
      <CustomLink to="/Calculator">Calculator</CustomLink>
      <CustomLink to="/Qoute">Qoute</CustomLink>
    </ul>
  </nav>
);

function CustomLink({ to, children, className = '' }) {
  const resolvedPath = useResolvedPath(to);
  const isActive = useMatch({ path: resolvedPath.pathname, end: true });
  return (
    <li className={isActive === to ? 'active' : ''}>
      <Link to={to} className={className}>
        {children}
      </Link>
    </li>
  );
}

CustomLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

CustomLink.defaultProps = {
  className: '',
};

export default Navbar;
