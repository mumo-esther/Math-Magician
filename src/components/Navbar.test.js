import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar from "./Navbar";

describe("Navbar component", () => {
  test("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("navigates to the correct page on click", () => {
    const { getByText } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    fireEvent.click(getByText("Calculator"));
    expect(window.location.pathname).toBe("/Calculator");
  });
});
