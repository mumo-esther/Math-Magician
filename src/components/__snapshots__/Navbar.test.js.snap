// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<nav
  class="nav"
>
  <a
    class="title"
    href="/"
  >
    Math Magicians
  </a>
  <ul>
    <li
      class=""
    >
      <a
        class=""
        href="/Home"
      >
        Home
      </a>
    </li>
    <li
      class=""
    >
      <a
        class=""
        href="/Calculator"
      >
        Calculator
      </a>
    </li>
    <li
      class=""
    >
      <a
        class=""
        href="/Qoute"
      >
        Qoute
      </a>
    </li>
  </ul>
</nav>
`;
